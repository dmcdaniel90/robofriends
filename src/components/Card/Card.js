import React, { useState } from "react";
import placeholder from '../../images/placeholder.jpg';

const Card = ({ name, email, id }) => {

  const robotImageUrl = `https://robohash.org/${id}?size=200x200`;
  const [hasImageLoaded, setImageLoaded] = useState(false);

  return ( 
    <div className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img src={hasImageLoaded ? robotImageUrl : placeholder} alt="robots" onLoad={() => setImageLoaded(true)} rel="preload" fetchpriority="high" width={"200px"} height={"200px"} data-testid="card"/>
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  )
}
export default Card;