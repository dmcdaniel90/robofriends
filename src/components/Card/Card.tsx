import React, { useState } from "react";
import placeholder from '../../images/placeholder.jpg';

type CardProps = {
  name: string,
  email: string,
  id: string
}

const Card = ({ name, email, id }: CardProps) => {

  const robotImageUrl: string = `https://robohash.org/${id}?size=200x200`;
  const [hasImageLoaded, setImageLoaded] = useState(false);

  return ( 
    <div className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img src={hasImageLoaded ? robotImageUrl : placeholder} alt="robots" onLoad={() => setImageLoaded(true)} width={"200px"} height={"200px"} data-testid="card"/>
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  )
}
export default Card;