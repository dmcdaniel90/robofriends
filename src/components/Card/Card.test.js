import {screen, render, fireEvent} from '@testing-library/react'
import Card from './Card';

describe('Card', () => {
  const mockProps = {
    name: 'John Doe',
    email: 'johndoe@gmail.com',
    id: 1
  }
  
  it('expect to render Card component', () => {
    const {container} = render(<Card {...mockProps} />)
    expect(container.firstChild).toMatchSnapshot();
  })

  it('renders the placeholder image when the image has not loaded', () => {
    render(<Card {...mockProps} />)
    const image = screen.getByTestId('card')

    expect(image).toHaveAttribute('src', 'placeholder.jpg')
  })

  it('renders the robot image when the image has loaded', () => {
    const id = 1

    render(<Card {...mockProps} />)
    const image = screen.getByTestId('card')

    fireEvent.load(image)
    expect(image).toHaveAttribute('src', `https://robohash.org/${id}?size=200x200`)
  })
})