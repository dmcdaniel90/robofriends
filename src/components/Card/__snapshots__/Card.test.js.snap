// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card expect to render Card component 1`] = `
<div
  class="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    data-testid="card"
    fetchpriority="high"
    height="200px"
    rel="preload"
    src="placeholder.jpg"
    width="200px"
  />
  <div>
    <h2>
      John Doe
    </h2>
    <p>
      johndoe@gmail.com
    </p>
  </div>
</div>
`;
